optuna/structs.py

```diff
@@ -126,9 +126,9 @@ def __init__(
     def __eq__(self, other):
         # type: (Any) -> bool
 
-        if isinstance(other, type(self)):
-            return other.__dict__ == self.__dict__
-        return False
+        if not isinstance(other, FrozenTrial):
```

Comment:
[nits] It looks duplicate with https://github.com/optuna/optuna/commit/9ec8949e64fd5d8c98e595c0cf7e28e41e949e21 at #718.

Reply:
Ah yes, it's an identical fix but for a different class so they're modifying different pieces of code.